enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,511 @@
+
+#### Failure to reboot
+
+It is possible that a single node may fail to reboot due to an issue
+with configuration, hardware, or environment. These single-node
+deployments will be managed by external tools, and as with other
+failures related to providing resilient operation, host boot failures
+will need to be detected and mitigated by those tools, rather than the
+local OpenShift instance.
+
+#### Reboots for reconfiguration
+
+Some reconfiguration operations will unavoidably cause the node to
+reboot. These tasks will need to be scheduled during maintenance
+windows.  Ideally we will be able to rely on the [graceful shutdown
+support in
+kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown)
+to stop all workloads safely as part of the reboot.
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+2. How do we hand off ownership of the `etcd-quorum-guard` Deployment
+   from the `cluster-version-operator` to the `cluster-etcd-operator`?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have CI
+coverage informing the release. An end-to-end job using a single-node
+deployment and running an appropriate subset of the standard OpenShift
+tests will be created and configured to block accepting release images
+unless it passes.
+
+That end-to-end job should also be run against pull requests for the
+operators and other components that are most affected by the new
+configuration, such as the etcd and auth operators.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+    - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+    - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
```

Comment:
can we add a dev preview section with a sketch of what is to be delivered?

I'd also expect that some test coverage is added here.

Reply:
also mention that upgrade is not a deliverable

Reply:
Upgrades are mentioned in the non-goals section on line 158.

Test coverage has its own section, so I'm not sure what additional details belong here. What sort of thing did you have in mind?

The rest of this section is still just the template content, and could stand to be fleshed out.

Reply:
so to be clear in graduation criteria you add something for each stage that the feature is relased (usually alpha, beta, GA) in this case we have tech preview and GA but nothing for dev preview, i'd expect to see a section showing deliverable for dev preview (at least enough to differentiate between it and tech preview).

Reply:
I have updated the various criteria based on my current understanding.

Reply:
thank you!